tests/login.go

```diff
@@ -19,7 +19,9 @@ import (
 // LoginToCirros call LoggedInFedoraExpecter but does not return the expecter
 func LoginToCirros(vmi *v1.VirtualMachineInstance) error {
 	expecter, err := LoggedInCirrosExpecter(vmi)
-	defer expecter.Close()
+	if err == nil {
+		expecter.Close()
+	}
```

Comment:
This is correct, but is a bit unsafe if changes to this helper occur in the future.

The preferred pattern is this: If the operation failed, just return early, no need to do anything else. If it is not an error, defer the close.
This way, any changes to this function in the future will not risk forgetting to close the expecter. 

Sorry if I mislead you before or did not clarify this point.

Reply:
I don't mind either way. However since this function is solely there to succeed and then close, I find a `defer` confusing. It kind of implies, `let's close later after we are done with it` but we don't want to do anything with it. @fgimenez both ways are ok for me. Whatever you prefer.

Reply:
@rmohr agree, as I see it defer would not add any value here.

Reply:
Defer will sign-off the function in a way that it is safe for editation. If you always use the same pattern to close resources the chances for surprises and mistakes are reduced.

But I will not insist here.